### ISSUE TITLE
Database adapter, PostgreSQL support 

### ISSUE THREAD

**Issue by avelino** [pydio/cells#85]

**Comment by bsinou** [comment]
Hi, 

We have been working on a first PoC a few weeks ago, (see branch `pgsql-poc`) and had been able to start Cells with service `pydio.grpc.acl` backed by a PostgreSQL DB.

This helped us to have a better overview of what has to be done and there is some work at hand, that is mainly:
- rewrite all DB creation / migration scripts for each microservice that is backed by a DB with PGSQL specific syntax 
- adapt our store procedures that are using MySQL specific place holders, for this we must either:
  - rewrite all parametrized stored procedure using Goqu
  - choose a strategy (Qoqu might also come handy here) to translate the stored procedures either once or dynamically during `init`phase of the service from MySQL syntax to PGSQL syntax: the place holder in prepared statements depend on the adapter (e.g. ? mysql, $1 postgres, ? sqlite3)
- enrich our non regression tests to insure switching to PGSQL does not break anything
- benchmark pretty big systems backed with PGSQL to insure it holds the load. This benchmarking would also help us to gather some best practices and PGSQL specific fine tuning tips.   

We do not really have the time to do this alone by ourselves in a very near future, yet we think this is a good candidate for a community driven effort to which we would take part. 

Would anyone be interested to take part in the process?

**Comment by LNCrafty** [comment]
Simillar to this Point: https://github.com/pydio/cells/issues/319.

Also did you guys heared from gorm ? Pretty good library for also handling multiple DB backends.
https://gorm.io/

**Comment by DUOLabs333** [comment]
The POC has not been updated in quite a while. Is it able to just be rebased on master?

**Comment by elhananjair** [comment]
Postgres is very popular now a days and easy to configure it would be best if it available as an option

**Comment by bsinou** [comment]
We are working on it. Stay tuned.

**Comment by aleksasiriski** [comment]
Any updates on this?

**Comment by cdujeu** [comment]
yep - it's going well in the current development cycle. Target for v5, ~mid november

**Comment by arana198** [comment]
Any update on v5?

**Comment by giesmininkas** [comment]
Good day!

Since v5 has now been released, with support for PostgreSQL database, how would one migrate from a MySQL-backed v4 instance to a v5 PostgreSQL-backed one?

Thank you for your great work!

**Comment by ghecquet** [comment]
Hi, this is not documented as yet, but you can migrate to v5 on the mysql database and run the following commands (pgloader and psql required)
Make sure you change the credentials and hosts in the migration.pdb file

```
pgload migration.pdb
psql 'postgresql://pydio:pydio123@127.0.0.1:5432/pydio' -f post-fix.sql
```

[migration.pdb](https://github.com/user-attachments/files/29837421/migration.pdb)
[post-fix.sql](https://github.com/user-attachments/files/29837420/post-fix.sql)

Closing this as the original issue is done
